src/batch/src/executor/postgres_query.rs

```diff
@@ -0,0 +1,190 @@
+        self.do_execute().boxed()
+    }
+}
+
+pub fn postgres_row_to_owned_row(
+    row: tokio_postgres::Row,
+    schema: &Schema,
+) -> Result<OwnedRow, BatchError> {
+    let mut datums = vec![];
+    for i in 0..schema.fields.len() {
+        let rw_field = &schema.fields[i];
+        let name = rw_field.name.as_str();
+        let datum = postgres_cell_to_scalar_impl(&row, &rw_field.data_type, i, name)?;
+        datums.push(datum);
+    }
+    Ok(OwnedRow::new(datums))
+}
+
+// TODO(kwannoel): Support more types, see postgres connector's ScalarAdapter.
+fn postgres_cell_to_scalar_impl(
+    row: &tokio_postgres::Row,
+    data_type: &DataType,
+    i: usize,
+    name: &str,
+) -> Result<Datum, BatchError> {
+    let datum = match data_type {
+        DataType::Boolean
+        | DataType::Int16
+        | DataType::Int32
+        | DataType::Int64
+        | DataType::Float32
+        | DataType::Float64
+        | DataType::Date
+        | DataType::Time
+        | DataType::Timestamp
+        | DataType::Timestamptz
+        | DataType::Jsonb
+        | DataType::Interval
+        | DataType::Varchar
+        | DataType::Bytea => {
+            // ScalarAdapter is also fine. But ScalarImpl is more efficient
+            row.try_get::<_, Option<ScalarImpl>>(i)?
+        }
+        DataType::Decimal => {
+            // Decimal is more efficient than PgNumeric in ScalarAdapter
+            let val = row.try_get::<_, Option<Decimal>>(i)?;
+            val.map(ScalarImpl::from)
+        }
+        _ => {
+            tracing::warn!(name, ?data_type, "unsupported data type, set to null");
+            None
+        }
+    };
+    Ok(datum)
+}
+
+impl PostgresQueryExecutor {
+    pub fn new(
+        schema: Schema,
+        host: String,
+        port: String,
+        username: String,
+        password: String,
+        database: String,
+        query: String,
+        identity: String,
+    ) -> Self {
+        Self {
+            schema,
+            host,
+            port,
+            username,
+            password,
+            database,
+            query,
+            identity,
+        }
+    }
+
+    #[try_stream(ok = DataChunk, error = BatchError)]
+    async fn do_execute(self: Box<Self>) {
+        tracing::debug!("postgres_query_executor: started");
+        let conn_str = format!(
+            "host={} port={} user={} password={} dbname={}",
+            self.host, self.port, self.username, self.password, self.database
+        );
+        let (client, conn) = tokio_postgres::connect(&conn_str, tokio_postgres::NoTls).await?;
+
+        tokio::spawn(async move {
+            if let Err(e) = conn.await {
+                tracing::error!(
+                    "postgres_query_executor: connection error: {:?}",
+                    e.as_report()
+                );
+            }
+        });
+
+        // TODO(kwannoel): Use pagination using CURSOR.
+        let rows = client.query(&self.query, &[]).await?;
```

Comment:
```suggestion
        let rows = client.query(&self.query, &[]).await.context("postgres_query received error from remote server")?;
```

Without this additional context, it looks to be a RisingWave error:
```
dev=> select * from postgres_query('localhost', '5432', 'postgres', 'postgres', 'test', 'select 2147483647/0 from z');
ERROR:  Failed to run the query

Caused by these errors (recent errors listed first):
  1: db error
  2: ERROR: division by zero
```